### PR TITLE
BF: Correct assertion to check what we are actually interested in.

### PR DESCRIPTION
**Pull request by bpoldrack**
See #1319

### CODE REVIEW

File: datalad/distribution/tests/test_publish.py [hunk 1]
```diff
@@ -95,8 +95,9 @@ def test_publish_simple(origin, src_path, dst_path):
     ok_clean_git(dst_path, annex=None)
     eq_(list(target.get_branch_commits("master")),
         list(source.repo.get_branch_commits("master")))
-    eq_(list(target.get_branch_commits("git-annex")),
-        list(source.repo.get_branch_commits("git-annex")))
+    assert(all(
+        [commit in list(target.get_branch_commits("git-annex"))
+         for commit in list(source.repo.get_branch_commits("git-annex"))]))
```
Comment (yarikoptic) [inline datalad/distribution/tests/test_publish.py]: I would have used then set not a list and it's  .difference

**Comment by bpoldrack** [comment]
Better, @yarikoptic ?

**Comment by yarikoptic** [comment]
"good enough" ;)   I would have also added a comment about this so later we don't wonder wtf... will do directly after merge

**Comment by yarikoptic** [comment]
btw with this -- uploaded that new annex to stock neurodebian @bpoldrack

**Pull request merged**